x-pack/qa/sql/src/main/java/org/elasticsearch/xpack/qa/sql/jdbc/DocsCsvSpecTestCase.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.qa.sql.jdbc;
+
+import com.carrotsearch.randomizedtesting.annotations.ParametersFactory;
+
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.xpack.qa.sql.jdbc.CsvTestUtils.CsvTestCase;
+import org.elasticsearch.xpack.sql.jdbc.jdbc.JdbcConfiguration;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Properties;
+
+import static org.elasticsearch.xpack.qa.sql.jdbc.CsvTestUtils.csvConnection;
+import static org.elasticsearch.xpack.qa.sql.jdbc.CsvTestUtils.executeCsvQuery;
+import static org.elasticsearch.xpack.qa.sql.jdbc.CsvTestUtils.specParser;
+
+/**
+ * CSV test specification for DOC examples.
+ * While we could use the existing tests, their purpose is to test corner-cases which
+ * gets reflected in the dataset structure.
+ * The doc tests while redundant, try to be expressive first and foremost and sometimes
+ * the dataset isn't exactly convenient.
+ * 
+ * Also looking around for the tests across the test files isn't trivial.
+ * 
+ * That's not to say the two cannot be merged however that felt like too much of an effort
+ * at this stage and, to not keep things stalling, started with this approach.
+ */
+public abstract class DocsCsvSpecTestCase extends SpecBaseIntegrationTestCase {
+
+    private final CsvTestCase testCase;
+
+
+    @Override
+    protected String indexName() {
+        return "library";
+    }
+
+    @Override
+    protected void loadDataset(RestClient client) throws Exception {
+        DataLoader.loadDocsDatasetIntoEs(client);
+    }
+
+    @ParametersFactory(shuffle = false, argumentFormatting = SqlSpecTestCase.PARAM_FORMATTING)
+    public static List<Object[]> readScriptSpec() throws Exception {
+        Parser parser = specParser();
+        return readScriptSpec("/docs.csv-spec", parser);
+    }
+
+    public DocsCsvSpecTestCase(String fileName, String groupName, String testName, Integer lineNumber, CsvTestCase testCase) {
+        super(fileName, groupName, testName, lineNumber);
+        this.testCase = testCase;
+    }
+
+    @Override
+    protected void assertResults(ResultSet expected, ResultSet elastic) throws SQLException {
+        Logger log = logEsResultSet() ? logger : null;
+
+        //
+        // uncomment this to printout the result set and create new CSV tests
+        //
+        JdbcTestUtils.logLikeCLI(elastic, log);
+        //JdbcAssert.assertResultSets(expected, elastic, log);
```

Comment:
This look like it isn't asserting anything. I don't mind the printing being enabled all the time but I think we should have the assert on, right?

Reply:
You're right - the comment will disappear in the final commit. I'm printing stuff out to help with the docs (logging while asserting adds the log header which makes it harder to copy-paste the text output).